libs/typescript/core/src/clients/artifacts/index.ts

```diff
@@ -1,26 +1,41 @@
 import { ArtifactsClient } from './base';
 import { DatabricksArtifactsClient } from './databricks';
 import { MlflowArtifactsClient } from './mlflow';
+import { AuthProvider } from '../../auth';
+
+/**
+ * Options for creating an ArtifactsClient.
+ */
+export interface ArtifactsClientOptions {
+  /**
+   * The tracking URI (e.g., "databricks", "http://localhost:5000")
+   */
+  trackingUri: string;
+
+  /**
+   * The resolved host URL for API requests
+   */
+  host: string;
+
+  /**
+   * Authentication provider
+   */
+  authProvider: AuthProvider;
+}
 
 /**
  * Get the appropriate artifacts client based on the tracking URI.
  *
- * @param trackingUri - The tracking URI to use to determine the artifacts client.
+ * @param options - Options for creating the artifacts client
  * @returns The appropriate artifacts client.
  */
-export function getArtifactsClient({
-  trackingUri,
-  host,
-  databricksToken
-}: {
-  trackingUri: string;
-  host: string;
-  databricksToken?: string;
-}): ArtifactsClient {
+export function getArtifactsClient(options: ArtifactsClientOptions): ArtifactsClient {
```

Comment:
Let's keep using destructuring assignment. We can use destructuring assignment and interface together.